nova/core/roles/customization_single_role/tasks/main.yml

```diff
@@ -1,35 +1,26 @@
 ---
-- name: Checking if the local role exists...
+- name: Checking if {{ single_role }} exists in {{ customization_single_role_local_root_folder }}/{{ single_role }}...
   ansible.builtin.stat:
-    path: "{{ inventory_dir }}/roles/{{ single_role }}"
+    path: "{{ customization_single_role_local_root_folder }}/{{ single_role }}"
   register: local_role_existence
   become: false
   delegate_to: localhost
 
-- name: Checking if the shared role exists...
-  ansible.builtin.stat:
-    path: "{{ single_role }}"
-  register: shared_role_existence
-  become: false
-  delegate_to: localhost
-
-- name: Making sure that {{ post_scripts_folder_on_target }} folder exists for Windows...
+- name: Making sure that {{ post_scripts_folder_on_target }} folder exists...
```

Comment:
I think for this task we still need to check if the shared_role exist and then add to this task another when: `local_role_existence.stat.exists or shared_role_existence.stat.exists`

Maybe same should be config in `nova/core/roles/customization_pre_role/tasks/main.yml`